src/dvclive/plots/annotations.py

```diff
@@ -0,0 +1,166 @@
+from typing import List, Literal, Union, Dict, get_args
+import math
+
+import numpy as np
+from pathlib import Path
+
+from dvclive.plots.utils import NumpyEncoder
+from dvclive.serialize import dump_json
+import logging
+
+from .base import Data
+
+
+logger = logging.getLogger("dvclive")
+
+BOXES_NAME = "boxes"
+LABELS_NAME = "labels"
+SCORES_NAME = "scores"
+FORMAT_NAME = "format"
+
+BboxFormatKind = Literal["tlbr", "tlhw", "xywh", "ltrb"]
+
+
+class Annotations(Data):
+    suffixes = (".json",)
+    subfolder = "images"
+
+    @property
+    def output_path(self) -> Path:
+        _path = self.output_folder / self.name
+        _path.parent.mkdir(exist_ok=True, parents=True)
+        return _path
+
+    @staticmethod
+    def could_log(  # noqa: PLR0911
+        annotations: Dict[str, List],
+    ) -> bool:
+        # no missing fields
+        if any(
+            field not in annotations
```

Comment:
can we use some schema parser like https://pypi.org/project/voluptuous/ ? do other project validate the schema?

Reply:
@skshetry what are the options for validating a schema like this with some meaningful error and preferably with the least amount of dependenices?

Reply:
`voluptuous` is not being maintained but has no dependencies.

I'd use `pydantic` as it works with type-hints and has dataclass-like interface and is more popular and has good DX.

Both `pydantic` and `voluptuous` are dependencies/transitive-dependencies of dvc, so I'd not worry about dependencies.